drakrun/drakrun/vm.py

```diff
@@ -91,26 +92,80 @@ def is_running(self) -> bool:
         )
         return res.returncode == 0
 
-    def restore(self) -> None:
+    def create(self, cfg_path, pause=False, **kwargs):
+        args = ['xl', 'create']
+        if pause:
+            args += ['-p']
+        args += [cfg_path]
+        logging.info(f"Creating VM {self.vm_name}")
+        try_subprocess(args, f"Failed to launch VM {self.vm_name}", **kwargs)
+
+    def pause(self, **kwargs):
+        logging.info(f"Pausing VM {self.vm_name}")
+        try_subprocess(['xl', 'pause', self.vm_name], f"Failed to pause VM {self.vm_name}", **kwargs)
+
+    def unpause(self, **kwargs):
+        logging.info(f"Unpausing VM {self.vm_name}")
+        try_subprocess(['xl', 'unpause', self.vm_name], f"Failed to unpause VM {self.vm_name}", **kwargs)
+
+    def save(self, filename, pause=False, cont=False, **kwargs):
+        logging.info(f"Saving VM {self.vm_name}")
+        args = ['xl', 'save']
+
+        # no such args will shutdown the VM after saving
+        if pause is True:
+            args += ['-p']
+        elif cont is True:
+            args += ['-c']
+
+        if kwargs.get('stderr') is None:
+            kwargs['stderr'] = kwargs['stdout'] = subprocess.STDOUT
+
+        args += [self.vm_name, filename]
+
+        try_subprocess(args, f"Failed to save VM {self.vm_name}", **kwargs)
+
+    def restore(
+        self,
+        cfg_path=None,
```

Comment:
I don't think that this should be exposed. Configuration is bound to the identity of the VM, overriding the config path is somewhat risky.

Adding snapshot path is OK because snapshots of the same VM are interchangeable.

Reply:
hmm, makes sense, so how do you propose I pass the configuration during `VM_tests`? Coz we don't want the original configurations to be used that time and I was passing it through this argument. 

During the tests, we get a temporary file name which stores the config throughout the period of the test and later gets deleted, So, i don't think I'll be able to monkeypatch the cfg_path internally

Reply:
What do you think about moving the `cfg_path` from restore() and create() to the constructor?
The default would be to use `cfg_path = Path(VM_CONFIG_DIR) / f"{self.vm_name}.cfg"`, but there would be an option to set it manually

Reply:
> What do you think about moving the `cfg_path` from restore() and create() to the constructor?
> The default would be to use `cfg_path = Path(VM_CONFIG_DIR) / f"{self.vm_name}.cfg"`, but there would be an option to set it manually

Sounds nice to me